sklearn/linear_model/sag.py

```diff
@@ -193,31 +204,34 @@ def sag_solver(X, y, sample_weight=None, loss='log', alpha=1.,
     if 'coef' in warm_start_mem.keys():
         coef_init = warm_start_mem['coef']
     else:
-        coef_init = np.zeros(n_features, dtype=np.float64, order='C')
+        # assume fit_intercept is False
+        coef_init = np.zeros((n_features, n_classes), dtype=np.float64,
+                             order='C')
 
     # coef_init contains possibly the intercept_init at the end.
     # Note that Ridge centers the data before fitting, so fit_intercept=False.
-    fit_intercept = coef_init.size == (n_features + 1)
+    fit_intercept = coef_init.shape[0] == (n_features + 1)
     if fit_intercept:
-        intercept_init = coef_init[-1]
-        coef_init = coef_init[:-1]
+        intercept_init = coef_init[-1, :]
+        coef_init = coef_init[:-1, :]
     else:
-        intercept_init = 0.0
+        intercept_init = np.zeros(n_classes, dtype=np.float64)
 
     if 'intercept_sum_gradient' in warm_start_mem.keys():
-        intercept_sum_gradient_init = warm_start_mem['intercept_sum_gradient']
+        intercept_sum_gradient = warm_start_mem['intercept_sum_gradient']
```

Comment:
Can you document `warm_start_mem`? Right now I don't know what they are without looking into the code.